src/Conversion/KrnlToLLVM/KrnlToLLVM.cpp

```diff
@@ -375,80 +375,86 @@ class KrnlGlobalOpLowering : public ConvertToLLVMPattern {
           StringAttr llvmStringAttr = StringAttr::get(data, context);
           global = rewriter.create<LLVM::GlobalOp>(loc, llvmArrayI8Ty,
               /*isConstant=*/true, LLVM::Linkage::Internal, name,
               llvmStringAttr);
-        } else if (krnlGlobalOp.value().getValue().isa<DenseElementsAttr>()) {
-          DenseElementsAttr denseAttr =
-              krnlGlobalOp.value().getValue().cast<DenseElementsAttr>();
-          if (!denseAttr.isSplat()) {
-            std::vector<char> rawData = denseAttr.getRawData();
-            // Check data size.
-            assert((rawData.size() == sizeInBytes) && "Data size mismatch.");
-
-            StringRef data = StringRef((char *)rawData.data(), rawData.size());
-            StringAttr llvmStringAttr = StringAttr::get(data, context);
-            global = rewriter.create<LLVM::GlobalOp>(loc, llvmArrayI8Ty,
-                /*isConstant=*/true, LLVM::Linkage::Internal, name,
-                llvmStringAttr);
-          } else {
-            global = rewriter.create<LLVM::GlobalOp>(loc, llvmGlobalType,
-                /*isConstant=*/true, LLVM::Linkage::Internal, name,
-                krnlGlobalOp.value().getValue());
-          }
-        } else
-          llvm_unreachable("Unsupported attribute type");
-      }
+        } else {
+          global = rewriter.create<LLVM::GlobalOp>(loc, llvmGlobalType,
+              /*isConstant=*/true, LLVM::Linkage::Internal, name,
+              krnlGlobalOp.value().getValue());
+        }
+      } else
+        llvm_unreachable("Unsupported attribute type");
+    }
 
+    // Prepare data to be inserted into MemRef.
+    // If global needs alignment, we need to allocate a new buffer that is
+    // aligned. This is a region of local memory and needs to be emitted as an
+    // alloca.
+    // - Otherwise, no preparation is needed, directly use the global.
+
+    int alignment = 0;
+    if (krnlGlobalOp.alignmentAttr())
+      alignment = krnlGlobalOp.alignmentAttr().getValue().getSExtValue();
+    Value typedGlobal;
+
+    // Get the address of the global.
+    Value globalValue = rewriter.create<LLVM::AddressOfOp>(loc, global);
+    auto llvmConstantElementType = constantElementType.cast<Type>();
+
+    if (alignment != 0) {
       // Some frequently used types.
       auto llvmI8PtrTy =
           LLVM::LLVMPointerType::get(IntegerType::get(context, 8));
       auto llvmI64Ty = IntegerType::get(context, 64);
-
-      // Allocate the memory where the constants will be used from.
-      // This is a region of local memory and needs to be emitted as an alloca.
-      int alignment = 0;
-      if (krnlGlobalOp.alignmentAttr())
-        alignment = krnlGlobalOp.alignmentAttr().getValue().getSExtValue();
-
       auto one = rewriter.create<LLVM::ConstantOp>(
           loc, llvmI64Ty, rewriter.getI64IntegerAttr(1));
-      alloc = rewriter.create<LLVM::AllocaOp>(loc,
+      Value alloc = rewriter.create<LLVM::AllocaOp>(loc,
```

Comment:
This comment is not specific to this patch, but it seems like allocating parameters for weights on the stack is a risky idea, given the possibility of stack overflow. Plus weight tensors are pretty big.